ambry-network/src/main/java/com/github/ambry/network/http2/Http2NetworkClient.java

```diff
@@ -124,8 +164,11 @@ public void operationComplete(ChannelFuture future) throws Exception {
                     releaseAndCloseStreamChannel(streamChannel);
                     http2ClientResponseHandler.getResponseInfoQueue()
                         .put(new ResponseInfo(requestInfo, NetworkClientErrorCode.NetworkError, null));
-                    // release related bytebuf
-                    requestInfo.getRequest().release();
+                    if (!(future.cause() instanceof ClosedChannelException)) {
+                      // If it's ClosedChannelException caused by drop request, it's probably refCnt has been decreased.
```

Comment:
Why doesn't the ref count get decremented in other exception cases? Are we assuming that other exception cases come from ChannelHandlers written by Ambry that don't decrement the ref count?

Reply:
It's because I am not clear when exactly refCnt is decreased. Let me do more reading.

Reply:
@cgtz @justinlin-linkedin   I need suggestions from you.

I just did an experiment which added  an excpetion throw in AmbrySendToHttp2Adaptor.write() (after `ctx.write(dataFrame, promise)`). In this case, writeAndFlush listener see refCnt 1. 

But in the other case we saw, channel closed, writeAndFlush listener is also triggered. refCnt is 0. (because if we do one more refCnt decrase, netty reports error.)